### ISSUE TITLE
Some doubts about the paper and code

### ISSUE THREAD

**Issue by XiaoHaoPan** [nihaomiao/CVPR2023_LFDM#25]
1.What are the three losses in the first stage?

![image](https://github.com/nihaomiao/CVPR23_LFDM/assets/127167436/2047de82-0279-4778-9885-0309fbf92af5)

2.The generator model in this parameter seems to contain three trainable networks in one stage, is it correct?

![image](https://github.com/nihaomiao/CVPR23_LFDM/assets/127167436/748283fb-a761-4ead-aa04-0aaf1bfd6f13)

3.What is the purpose of adding x0 in the second stage, I did not find it in the paper

![image](https://github.com/nihaomiao/CVPR23_LFDM/assets/127167436/4e28d19f-4509-4767-b24f-6873b900f00f)

4.I am not very clear about the division of test sets and datasets

**Comment by nihaomiao** [comment]
Hi, @XiaoHaoPan,

1. they are perceptual loss and equivariance loss. You may refer to Section 4.2 Model Implementation in our paper.
2. Correct.
3. We use $x_0$ to provide the shape information for the flow and occlusion map.
4. The split of the train/test set can be found in our preprocessing codes.

**Comment by XiaoHaoPan** [comment]
Thank you for answering, I have another question, please ask the video generated in the first phase of the test, where the second frame generated the frame, the reference frame used is the first real driver frame, or the model generated fake driver frame？

**Comment by XiaoHaoPan** [comment]
Please ask how the settings in your paper are different from those in the code. Do I need to change the epoch settings to achieve the pre-training effect you provide?

![image](https://github.com/nihaomiao/CVPR23_LFDM/assets/127167436/66299a69-945a-4aa9-b722-846038dcc31d)
![image](https://github.com/nihaomiao/CVPR23_LFDM/assets/127167436/f6ca5186-62d3-405d-8d4e-ea8138271b28)

**Comment by nihaomiao** [comment]
Hi, @XiaoHaoPan, during testing, we use DM to generate a sequence of flow maps and apply flow maps to the given real reference image to generate a sequence of generated frames. We always keep warping the real reference frame to maintain good quality.

**Comment by XiaoHaoPan** [comment]
Thanks！I looked at your training DM phase code and I found that these two settings are different for different datasets.
Is it because of the differences in different datasets?

MUG:
![image](https://github.com/nihaomiao/CVPR23_LFDM/assets/127167436/b6b6f7ed-0f8b-4250-91c9-d594d10a0154)

MHAD:
![image](https://github.com/nihaomiao/CVPR23_LFDM/assets/127167436/1ecb67e0-2e99-4129-91d8-94819ac7b070)

NATOPS:
![image](https://github.com/nihaomiao/CVPR23_LFDM/assets/127167436/243867f6-cc75-41e2-9e8e-63c1d4db393b)

Did the pre-trained model you provided train with this setup?

**Comment by nihaomiao** [comment]
Hi, @XiaoHaoPan, Sorry for misunderstanding your second question! I have updated my previous comments. Yes, the size of NATOPS is much larger than the MUG and MHAD. So I set a smaller number of epochs. The pre-trained models I provided should be consistent with these settings if I didn't edit the code later. But it has been several months ago and I am not very sure about that.
